### PR TITLE
Task reschedule

### PR DESCRIPTION
**Pull request by r0xsh**
https://github.com/coopcycle/coopcycle-web/assets/5613584/b1d3d524-21c8-4a9b-bc64-06574516d0a0

### CODE REVIEW

File: src/Domain/Task/Handler/RescheduleHandler.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public function __invoke(Reschedule $command)
+    {
+        $task = $command->getTask();
+        $rescheduleDateTime = $command->getRescheduleDateTime();
+
+        $this->eventRecorder->record(new Event\TaskRescheduled($task, $rescheduleDateTime));
+
+        $task->setDoneBefore($rescheduleDateTime);
+        $task->setMetadata('rescheduled', true);
```
Comment (r0xsh) [inline src/Domain/Task/Handler/RescheduleHandler.php:24]: i'm unsure about the metadata usage, maybe using a `Tags` "rescheduled" on the `Task` is better ? @alexsegura

Comment (r0xsh) [reply src/Domain/Task/Handler/RescheduleHandler.php:24]: I'm also unsure if `setDoneBefore()` is the right attribute to update, seems like is work from my first tries.

**Comment by r0xsh** [comment]
Next iteration
- https://github.com/coopcycle/coopcycle-web/issues/3767

File: src/Domain/Task/Handler/RescheduleHandler.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        $task = $command->getTask();
+        $rescheduledAfter = $command->getRescheduleAfter();
+        $rescheduledBefore = $command->getRescheduledBefore();
+
+        $this->eventRecorder->record(new Event\TaskRescheduled($task, $rescheduledAfter, $rescheduledBefore));
+
+        $task->setAfter($rescheduledAfter);
+        $task->setBefore($rescheduledBefore);
+        $task->unassign();
```
Comment (alexsegura) [inline src/Domain/Task/Handler/RescheduleHandler.php:26]: I'm not sure this makes sense... just in terms of UI, the task will "jump" from the assigned panel to the unassigned panel, and it's easy to lose track of the task when dealing with a dashboard full of tasks.

**Review by alexsegura — APPROVED** [review]

**Pull request merged**